### PR TITLE
Better handling of depot chest and lockers

### PR DESCRIPTION
**Pull request by ramon-bernardo**
### Pull Request Prelude

- [x] I have followed [proper The Forgotten Server code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

#### PR BASE #3698

[cont]: https://github.com/otland/forgottenserver/wiki/Contributing
[code]: https://github.com/otland/forgottenserver/wiki/TFS-Coding-Style-Guide

### CODE REVIEW

**Review by EPuncker — DISMISSED** [review]

File: src/depotchest.h [hunk 1]
```diff
@@ -29,8 +36,14 @@ class DepotChest final : public Container
 	Cylinder* getParent() const override;
 	Cylinder* getRealParent() const override { return parent; }
 
+	Player* getOwner() const { return owner; }
+	void setOwner(Player* owner) { this->owner = owner; }
```
Comment (ramon-bernardo) [inline src/depotchest.h:40]: Its not necess

File: src/player.cpp [hunk 2]
```diff
@@ -878,6 +878,7 @@ DepotChest* Player::getDepotChest(uint32_t depotId, bool autoCreate)
 
 	it = depotChests.emplace(depotId, new DepotChest(depotItemId)).first;
 	it->second->setMaxDepotItems(getMaxDepotItems());
+	it->second->setDepotId(depotId);
```
Comment (nekiro) [inline src/player.cpp:881]: what about moving this to constructor?

Comment (ramon-bernardo) [reply src/player.cpp]: Thanks!

File: src/depotchest.cpp [hunk 3]
```diff
@@ -7,8 +7,8 @@
 
 #include "tools.h"
 
-DepotChest::DepotChest(uint16_t type, bool paginated /*= true*/) :
-    Container{type, items[type].maxItems, true, paginated}
+DepotChest::DepotChest(uint16_t type, uint16_t depotId, bool paginated /*= true*/) :
+    Container{type, items[type].maxItems, true, paginated}, depotId(depotId)
```
Comment (ranisalt) [inline src/depotchest.cpp:11]:
```suggestion
    Container{type, items[type].maxItems, true, paginated}, depotId{depotId}
```

**Review by ranisalt — CHANGES_REQUESTED** [review]

**Pull request closed without merging**